test/test_codec.py

```diff
@@ -24,175 +27,175 @@ def test_bitstruct_format(self):
 
-    def test_encode_canmatrix(self):
-        db_path = os.path.join(
-            os.path.dirname(__file__), "..", "test", "test.dbc")
-        for bus in formats.loadp(db_path).values():
-            test_frame1 = 0x123
-            data = {
-                'Signal': 2,
-                'someTestSignal': 101,
-            }
-            data_bytes = tuple(bytearray(bus.encode(test_frame1, data)))
-            assert data_bytes == (0, 0x28, 0x04, 0, 0, 0, 0, 0)
+    # def test_encode_canmatrix(self):
+    #     db_path = os.path.join(
+    #         os.path.dirname(__file__), "..", "test", "test.dbc")
+    #     for matrix in formats.loadp(db_path).values():
+    #         test_frame1 = 0x123
+    #         data = {
+    #             'Signal': 2,
+    #             'someTestSignal': 101,
+    #         }
+    #         encoded = matrix.encode(test_frame1, data)
+    #         self.assertEqual(bytearray(encoded).hex(), bytearray((0, 0x28, 0x04, 0, 0, 0, 0, 0)).hex())
 
     def test_decode_signal(self):
-        s1 = Signal('signal', signalSize=8)
+        s1 = Signal('signal', size=8)
         self.assertEqual(s1.raw2phys(1), 1)
         self.assertEqual(s1.raw2phys(127), s1.max)
         self.assertEqual(s1.raw2phys(-128), s1.min)
 
-        s2 = Signal('signal', signalSize=10, is_signed=False)
+        s2 = Signal('signal', size=10, is_signed=False)
         self.assertEqual(s2.raw2phys(10), 10)
         self.assertEqual(s2.raw2phys(s2.max), 1023)
         self.assertEqual(s2.raw2phys(s2.min), 0)
 
-        s3 = Signal('signal', signalSize=8, factor=2)
+        s3 = Signal('signal', size=8, factor=2)
         self.assertEqual(s3.raw2phys(5), 10)
         self.assertEqual(s3.raw2phys(127), s3.max)
         self.assertEqual(s3.raw2phys(-128), s3.min)
 
-        s4 = Signal('signal', signalSize=8, is_signed=False, factor=5)
+        s4 = Signal('signal', size=8, is_signed=False, factor=5)
         self.assertEqual(s4.raw2phys(2), 10)
         self.assertEqual(s4.raw2phys(255), s4.max)
         self.assertEqual(s4.raw2phys(0), s4.min)
 
-        s5 = Signal('signal', signalSize=8, offset=2)
+        s5 = Signal('signal', size=8, offset=2)
         self.assertEqual(s5.raw2phys(8), 10)
         self.assertEqual(s5.raw2phys(127), s5.max)
         self.assertEqual(s5.raw2phys(-128), s5.min)
 
-        s6 = Signal('signal', signalSize=8, is_signed=False, offset=5)
+        s6 = Signal('signal', size=8, is_signed=False, offset=5)
         self.assertEqual(s6.raw2phys(5), 10)
         self.assertEqual(s6.raw2phys(255), s6.max)
         self.assertEqual(s6.raw2phys(0), s6.min)
 
-        s7 = Signal('signal', signalSize=16, is_signed=False, factor=0.001)
+        s7 = Signal('signal', size=16, is_signed=False, factor=0.001)
         self.assertEqual(s7.raw2phys(65535), s7.max)
         self.assertEqual(s7.raw2phys(0), s7.min)
-        self.assertEqual(s7.raw2phys(50123), 50.123)
+        self.assertAlmostEqual(s7.raw2phys(50123), D(50.123))
```

Comment:
Shouldn't using decimal get us exact?  But, definitely don't `decimal.Decimal(1.23)` but rather `decimal.Decimal('1.23')` (note the latter is a string).  I see this is closed, but still a relevant FYI on using decimal.